src/Schema/Builder.php

```diff
@@ -154,31 +157,36 @@ public function dropAllTables()
     }
 
     /** @param string|null $schema Database name */
-    public function getTables($schema = null)
-    {
+    private function getCollectionRows(
+        $schema = null,
+        ?string $collectionType = null,
+        ?callable $aggregateStats = null,
+    ) {
         $db = $this->connection->getDatabase($schema);
         $collections = [];
 
         foreach ($db->listCollections() as $collectionInfo) {
             $collectionName = $collectionInfo->getName();
 
-            // Skip views, which don't support aggregate
-            if ($collectionInfo->getType() === 'view') {
+            if ($collectionType && $collectionInfo->getType() !== $collectionType) {
                 continue;
             }
 
-            $stats = $db->selectCollection($collectionName)->aggregate([
-                ['$collStats' => ['storageStats' => ['scale' => 1]]],
-                ['$project' => ['storageStats.totalSize' => 1]],
-            ])->toArray();
+            $options = $collectionInfo->getOptions();
+            $collation = $options['collation'] ?? [];
+
+            // Aggregation is not supported on views
+            $shouldAggregateStats = $aggregateStats !== null && $collectionInfo->getType() !== 'view';
+
+            $stats = $shouldAggregateStats ? $aggregateStats($db, $collectionName) : null;
```

Comment:
If `$aggregateStats` is set only when `$collectionType == 'collection'`, this condition is the same as `$aggregateStats !== null`:

```suggestion
            $stats = $aggregateStats ? $aggregateStats($db, $collectionName) : null;
```

Reply:
Yes, but then it expects this method to know about how it's called. If we do change it I think I'd prefer not to pass the callable at all and just conditionally aggregate based on the passed collection type (essentially like what was done before).

I think that probably makes sense. The reason I passed the optional callable is to avoid having conditionals in a generic-ish method but I guess we need the conditional one way or another 😅

Reply:
Removing the callable would make the code more straightforward.